test/integration/hds_integration_test.cc

```diff
@@ -27,100 +34,435 @@ class HdsIntegrationTest : public HttpIntegrationTest,
+      hds_config->add_grpc_services()->mutable_envoy_grpc()->set_cluster_name("hds_cluster");
       auto* hds_cluster = bootstrap.mutable_static_resources()->add_clusters();
       hds_cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
       hds_cluster->mutable_circuit_breakers()->Clear();
-      hds_cluster->set_name("hds_delegate");
+      hds_cluster->set_name("hds_cluster");
       hds_cluster->mutable_http2_protocol_options();
-      // Switch predefined cluster_0 to EDS filesystem sourcing.
-      // TODO(lilika): Remove eds dependency
       auto* cluster_0 = bootstrap.mutable_static_resources()->mutable_clusters(0);
       cluster_0->mutable_hosts()->Clear();
-      cluster_0->set_type(envoy::api::v2::Cluster::EDS);
-      auto* eds_cluster_config = cluster_0->mutable_eds_cluster_config();
-      eds_cluster_config->mutable_eds_config()->set_path(eds_helper_.eds_path());
     });
+
     HttpIntegrationTest::initialize();
-    hds_upstream_ = fake_upstreams_[0].get();
-    for (uint32_t i = 0; i < upstream_endpoints_; ++i) {
-      service_upstream_[i] = fake_upstreams_[i + 1].get();
-    }
+
+    // Endpoint connections
+    host_upstream_.reset(new FakeUpstream(0, FakeHttpConnection::Type::HTTP1, version_));
+    host2_upstream_.reset(new FakeUpstream(0, FakeHttpConnection::Type::HTTP1, version_));
   }
 
+  // Sets up a connection between Envoy and the management server.
   void waitForHdsStream() {
     AssertionResult result =
-        hds_upstream_->waitForHttpConnection(*dispatcher_, fake_hds_connection_);
+        hds_upstream_->waitForHttpConnection(*dispatcher_, hds_fake_connection_);
     RELEASE_ASSERT(result, result.message());
-    result = fake_hds_connection_->waitForNewStream(*dispatcher_, hds_stream_);
+    result = hds_fake_connection_->waitForNewStream(*dispatcher_, hds_stream_);
     RELEASE_ASSERT(result, result.message());
   }
 
-  void requestHealthCheckSpecifier() {
-    envoy::service::discovery::v2::HealthCheckSpecifier server_health_check_specifier;
-    server_health_check_specifier.mutable_interval()->set_nanos(500000000); // 500ms
+  // Envoy sends healthcheck messages to the endpoints
+  void healthcheckEndpoints(std::string cluster2 = "None") {
+    ASSERT_TRUE(host_upstream_->waitForHttpConnection(*dispatcher_, host_fake_connection_));
+    ASSERT_TRUE(host_fake_connection_->waitForNewStream(*dispatcher_, host_stream_));
+    ASSERT_TRUE(host_stream_->waitForEndStream(*dispatcher_));
+
+    EXPECT_STREQ(host_stream_->headers().Path()->value().c_str(), "/healthcheck");
+    EXPECT_STREQ(host_stream_->headers().Method()->value().c_str(), "GET");
+    EXPECT_STREQ(host_stream_->headers().Host()->value().c_str(), "anna");
 
-    hds_stream_->sendGrpcMessage(server_health_check_specifier);
-    // Wait until the request has been received by Envoy.
-    test_server_->waitForCounterGe("hds_delegate.requests", ++hds_requests_);
+    if (cluster2 != "None") {
```

Comment:
Maybe just use empty string instead of "None", that should also be an invalid cluster name and is more idiomatic C++ for an optional.